hibernate-reactive-core/src/main/java/org/hibernate/reactive/context/impl/VertxContext.java

```diff
@@ -50,9 +46,9 @@ static ContextInternal currentContext() {
     @Override
     @SuppressWarnings("unchecked")
     public <T> T get(Key<T> key) {
-        final ContextInternal context = currentContext();
+        final io.vertx.core.Context context = Vertx.currentContext();
         if ( context != null ) {
-            return (T) context.localContextData().get( key );
+            return (T) context.get( key );
```

Comment:
@Sanne why did you change this to use the main context instead of the local context?

Isn't that wrong for what we're using it for?

Reply:
yes it's wrong - it wasn't intentional, I wasn't aware of the "global context" and all tests passed fine.

Ironically my benchmarks also showed no issue, as we don't use the APIs which use the context. Sorry :(

Reply:
@Sanne, we all share in the blame for this one. I approved it, Davide merged it. I *certainly* should have noticed but I just didn't read the diff carefully enough.

It's embarrassing but that's a think that can happen with this stuff that's hard to test. Hopefully we can figure out some way to test it.